extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -586,6 +586,9 @@ export class AmpAdXOriginIframeHandler {
    * @param {boolean} inViewport
    */
   viewportCallback(inViewport) {
+    if (this.intersectionObserverHost_) {
+      this.intersectionObserverHost_.onViewportCallback(inViewport);
```

Comment:
We should add a series of tests to prevent regressions here if possible.

Reply:
sure thing. as this need a quick cherrypick, how about in a separate PR?